### ISSUE TITLE
Missing `total=False` on some TypedDict with boto3-stubs>=1.28.3

### ISSUE THREAD

**Issue by Wurstnase** [youtype/mypy_boto3_builder#205]
**Describe the bug**
Some `TypeDict`s are not `total=False` anymore. With `mypy-boto3-secretsmanger==1.28.3` the flag is gone in the TypeDef. 

**To Reproduce**

1. Install `mypy-boto3-secretsmanger==1.28.3`
2. Run `mypy`/`pyright` on the following code sample

```python
from mypy_boto3_secretsmanger.type_defs import RotationRulesTypeTypeDef

rules: RotationRulesTypeTypeDef = {"AutomaticallyAfterDays": 365}
...
```

**Actual output**

```
error: Missing keys ("Duration", "ScheduleExpression") for TypedDict "RotationRulesTypeTypeDef"  [typeddict-item]
```

**Expected output**

```
no error
```

**Comment by DominikVollandTNG** [comment]
Same issue seen with S3 for multiple TypedDicts, specifically `CORSRuleTypeDef`, `ServerSideEncryptionByDefaultTypeDef`, `ServerSideEncryptionRuleTypeDef`, `LifecycleRuleTypeDef`, `LoggingEnabledTypeDef`, `TopicConfigurationTypeDef`, `MetricsConfigurationTypeDef`.

**Comment by vemel** [comment]
Thank you for the report!

Looks like the issue is related to a new logic for marking output `TypedDict` shapes as total. However, some shapes are used both as input and output. The proper fix would be to generate two `TypedDicts` (e.g. `RotationRulesTypeTypeDef` and `RotationRulesTypeOutputTypeDef`).

I will take a look.

**Comment by vemel** [comment]
Fixed in `mypy_boto3_builder 7.14.7` release. Please let me know if it works as it should.

I also published new packages with the hotfix included:

- `mypy_boto3_cognito_idp_package 1.28.3.post1`
- `mypy_boto3_connect_package 1.28.3.post1`
- `mypy_boto3_datasync_package 1.28.3.post1`
- `mypy_boto3_dms_package 1.28.3.post1`
- `mypy_boto3_ec2_package 1.28.3.post1`
- `mypy_boto3_fsx_package 1.28.3.post1`
- `mypy_boto3_iam_package 1.28.3.post1`
- `mypy_boto3_mediatailor_package 1.28.3.post1`
- `mypy_boto3_personalize_package 1.28.3.post1`
- `mypy_boto3_proton_package 1.28.3.post1`
- `mypy_boto3_s3_package 1.28.3.post1`
- `mypy_boto3_sagemaker_package 1.28.3.post1`
- `mypy_boto3_secretsmanager_package 1.28.3.post1`
- `mypy_boto3_package 1.28.3.post1`
- `boto3_stubs_package 1.28.3.post1`
- `boto3_stubs_lite_package 1.28.3.post1`

**Comment by ecs-jnguyen** [comment]
@vemel 

The `post1` is a breaking change for us. I started getting this error after the post1 version has been pushed:

```
    from mypy_boto3_ec2.type_defs import SubnetTypeDef
ImportError: cannot import name 'SubnetTypeDef' from 'mypy_boto3_ec2.type_defs'
```

**Comment by vemel** [comment]
@ecs-jnguyen Makes sense. I came up with a non-conflicting solution. I will release it in a few hours. Also, I am going to release `post2` to affected packages.

**Comment by vemel** [comment]
@ecs-jnguyen Just released `mypy-boto3-ec2 1.28.3.post2` with the fix included. Please test it and let me know if it works as it should.

Also, many other packages were updated, check [7.15.0 release notes](https://github.com/youtype/mypy_boto3_builder/releases/tag/7.15.0).

**Comment by Wurstnase** [comment]
LGTM. Thanks for your fast response and fixes.
